central/complianceoperator/v2/report/manager/manager_impl.go

```diff
@@ -104,12 +115,15 @@ func (m *managerImpl) Stop() {
 }
 
 func (m *managerImpl) generateReport(req *reportRequest) {
-	m.mu.Lock()
-	defer m.mu.Unlock()
+	defer m.concurrencySem.Release(1)
 
 	// TODO ROX-24172: Implement business logic for querying, formatting and sending report over email
 	logging.Infof("Executing report request for scan config %q", req.scanConfig.GetId())
+
+	m.mu.Lock()
```

Comment:
I suppose we could use `defer m.mu.Unlock()` here since we pushed this to the bottom.